File: Module 8/api_script.py
import requests
import json

def fetch_all_posts():
    url = 'https://jsonplaceholder.typicode.com/posts'
    try:
        response = requests.get(url)
        response.raise_for_status()
        posts = response.json()
        print("All Posts:")
        print(json.dumps(posts, indent=2))
        if posts:
            print("\nFirst Post Title:", posts[0]['title'])
    except requests.exceptions.RequestException as e:
        print(f"Error fetching posts: {e}")

fetch_all_posts()

File: Module 8/api_script.js
async function fetchAllPosts() {
  const url = 'https://jsonplaceholder.typicode.com/posts';
  try {
    const response = await fetch(url);
    if (!response.ok) throw new Error(`HTTP error! Status: ${response.status}`);
    const posts = await response.json();
    console.log("All Posts:", posts);
    if (posts.length > 0) console.log("First Post Title:", posts[0].title);
  } catch (error) {
    console.error("Error fetching posts:", error);
  }
}

fetchAllPosts();
